src/components/ChatBotFooter/ChatBotFooter.tsx

```diff
@@ -46,18 +46,34 @@ const ChatBotFooter = ({
 	// handles options for bot
 	const { botOptions } = useBotOptions();
 
+	const renderFileAttachment = () => {
+		return (
+			<FileAttachmentButton inputRef={inputRef}
+				flow={flow} getCurrPath={getCurrPath} openChat={openChat}
+				getPrevPath={getPrevPath} handleActionInput={handleActionInput} injectMessage={injectMessage}
+				streamMessage={streamMessage}
+			/>
+		)
+	}
+
+	const renderEmojiPicker = () => {
+		return (<EmojiPicker inputRef={inputRef} textAreaDisabled={textAreaDisabled}/>)
+	}
+	
 	return (
 		<div style={botOptions.footerStyle} className="rcb-chat-footer-container">
 			<div className="rcb-chat-footer">
-				{!botOptions.fileAttachment?.disabled &&
-					<FileAttachmentButton inputRef={inputRef} flow={flow} getCurrPath={getCurrPath} openChat={openChat}
-						getPrevPath={getPrevPath} handleActionInput={handleActionInput} injectMessage={injectMessage}
-						streamMessage={streamMessage}
-					/>
-				}
-				{!botOptions.emoji?.disabled &&
-					<EmojiPicker inputRef={inputRef} textAreaDisabled={textAreaDisabled}/>
-				}
+				{botOptions.footer?.buttons?.map((element) => {
+					if(element === "file-attachment" && !botOptions.fileAttachment?.disabled){
+						return renderFileAttachment()
+					}
+					else if (element === "emoji-picker" && !botOptions.emoji?.disabled){
+						return renderEmojiPicker()
+					}
+					else if(element !== "file-attachment" && element !== "emoji-picker"){
```

Comment:
The current approach means that if I add an invalid key e.g. "file_attachment", it will actually render a string. That is not desirable. Instead, if it doesn't match any of the keywords, we should ensure that it is a custom JSX element before rendering it. Otherwise we skip it entirely.